File: performance/cpp/sources/AllocationAndAccess.cpp
//#include <iostream>
//#include <chrono>
//#include <fstream>
//
//using namespace std;
//
//void WriteNumberToFile(const string &filename, const double number) {
//    ofstream logFile(filename, ios::app);
//    if (!logFile.is_open()) {
//        cerr << "Error opening file: " << filename << endl;
//        return;
//    }
//
//    logFile << number << endl;
//
//    if (logFile.fail()) {
//        cerr << "Error writing to file: " << filename << endl;
//    }
//    logFile.close();
//}
//
//int main() {
//
//    int arrayStatic[100000];
//
//    // Măsurarea timpului pentru alocarea de memorie dinamică
//    auto startDynamic = std::chrono::high_resolution_clock::now();
//    int *arrayDynamic = new int[100000];
//    auto endDynamic = std::chrono::high_resolution_clock::now();
//    std::cout << "Timpul pentru alocarea de memorie dinamica: "
//              << std::chrono::duration_cast<std::chrono::microseconds>(endDynamic - startDynamic).count()
//              << " microsecunde" << std::endl;
//
//    // Accesul la memorie
//    auto startAccessStatic = std::chrono::high_resolution_clock::now();
//    for (int i = 0; i < 100000; ++i) {
//        arrayStatic[i] = i; // Acces la memorie statică
//    }
//    auto endAccessStatic = std::chrono::high_resolution_clock::now();
//    std::cout << "Timpul pentru accesul la memorie statica: "
//              << std::chrono::duration_cast<std::chrono::microseconds>(endAccessStatic - startAccessStatic).count()
//
//              << " microsecunde" << std::endl;
//
//    auto startAccessDynamic = std::chrono::high_resolution_clock::now();
//    for (int i = 0; i < 100000; ++i) {
//        arrayDynamic[i] = i; // Acces la memorie dinamică
//    }
//    auto endAccessDynamic = std::chrono::high_resolution_clock::now();
//    std::cout << "Timpul pentru accesul la memorie dinamica: "
//              << std::chrono::duration_cast<std::chrono::microseconds>(endAccessDynamic - startAccessDynamic).count()
//
//              << " microsecunde" << std::endl;
//
//    delete[] arrayDynamic;
//
//    WriteNumberToFile("DynamicAlloc_log.cpp.txt",
//                      chrono::duration_cast<chrono::microseconds>(endDynamic - startDynamic).count());
//    WriteNumberToFile("StaticAccess_log.cpp.txt",
//                      chrono::duration_cast<chrono::microseconds>(endAccessStatic - startAccessStatic).count()
//    );
//    WriteNumberToFile("DynamicAccess_log.cpp.txt",
//                      chrono::duration_cast<chrono::microseconds>(endAccessDynamic - startAccessDynamic).count()
//    );
//
//    return 0;
//}

File: performance/python/AllocationAndAccess.py
import time

def write_number_to_file(filename, number):
    try:
        with open(filename, 'a') as file:
            file.write(str(number) + '\n')
    except IOError as e:
        print(f"Error opening/writing to file {filename}: {e}")

# Measure time for memory allocation
start_dynamic = time.time()
array_dynamic = [0] * 100000
end_dynamic = time.time()
memory_allocation_duration = int((end_dynamic - start_dynamic) * 1e6)

# Measure time for memory access
start_access_static = time.time()
for i in range(100000):
    array_dynamic[i] = i
end_access_static = time.time()
memory_access_duration = int((end_access_static - start_access_static) * 1e6)

# Write durations to a file
write_number_to_file("Alloc_log.py.txt", memory_allocation_duration)
write_number_to_file("Access_log.py.txt", memory_access_duration)
print(f"Memory allocation duration: {memory_allocation_duration}")
print(f"Memory access duration: {memory_access_duration}")
